include/envoy/http/codec.h

```diff
@@ -146,6 +146,16 @@ class ConnectionCallbacks {
    */
   virtual void onGoAway() PURE;
 };
+/**
+ * HTTP/1.* Codec settings
+ */
+
+struct Http1Settings {
+  // Enable codec to handle CONNECT requests. This will enable forward/explicit proxy support on TLS connections
+  bool allow_connect_{false} ;
+  // Enable codec to parse absolute uris. This enables forward/explicit proxy support for non TLS traffic
+  bool allow_absolute_url_{false};
```

Comment:
The comments here call out that these should work for tls/non-tls but there's no enforcement in the codec filter.  Do we want to allow absolute_urls for both secure and insecure (I'd lean towards yes because weird things happen on the internet) or do you want to enforce, perhaps via connection_.ssl() checks?